### PR TITLE
Add Dropbox parser

### LINKED ISSUE #3964: Dropbox sync_history.db parser
The Dropbox desktop application stores recently synced (added/deleted) files in a sqlite3 database.

Schema
```
CREATE TABLE sync_history (event_type TEXT NOT NULL,file_event_type TEXT,direction TEXT,file_id TEXT,local_path TEXT,timestamp INTEGER NOT NULL, other_user INTEGER)
```

Database location:
- Linux: $HOME/.dropbox/instance1/sync_history.db
- Windows 10: $HOME/AppData/Local/Dropbox/instance1/sync_history.db

### PR DESCRIPTION
**Pull request by sydp**
## One line description of pull request

Adds a parser for the Dropbox sync_history.db Sqlite database.

## Description:

**Related issue (if applicable):** fixes #3964 

## Notes:
All contributions to Plaso undergo [code
review](https://github.com/log2timeline/l2tdocs/blob/main/process/Code%20review%20process.asciidoc). This makes sure
that the code has appropriate test coverage and conforms to the Plaso [style
guide](https://plaso.readthedocs.io/en/latest/sources/developer/Style-guide.html).

One of the maintainers will examine your code, and may request changes. Check off the items below in
order, and then a maintainer will review your code.

## Checklist:
  - [ ] Automated checks (Travis, Codecov, Codefactor )pass
  - [ ] No new [new dependencies](https://plaso.readthedocs.io/en/latest/sources/developer/Adding-a-new-dependency.html) are required or l2tdevtools has been updated
  - [ ] Reviewer assigned

### CODE REVIEW

File: data/formatters/generic.yaml [hunk 1]
```diff
@@ -96,6 +96,16 @@ short_message:
 - 'Layer ID: {layer_id}'
 ---
 type: 'basic'
+data_type: 'dropbox:sync_history:entry'
+message:
+- 'File: {local_path}'
+- 'ID: {file_id}'
```
Comment (joachimmetz) [inline data/formatters/generic.yaml:102]: ID => identifier

File: plaso/parsers/sqlite_plugins/dropbox.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+
+class DropboxSyncHistoryEventData(events.EventData):
+  """Dropbox Sync History Database event data.
+
+  Attributes:
+    event_type (str): the event type
+    file_event_type (str): the file event type
+    direction (str): the source of the synchronisation event
+    file_id (str): the ID of the file.
```
Comment (joachimmetz) [inline plaso/parsers/sqlite_plugins/dropbox.py:20]: id => identifier

Also what does this identifier contain? is this a dropbox identifier?

File: plaso/parsers/sqlite_plugins/dropbox.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    self.file_event_type = None
+    self.direction = None
+    self.file_id = None
+    self.local_path = None
+
+
+class DropboxSyncDatabasePlugin(interface.SQLitePlugin):
+  """SQLite parser plugin for Dropbox sync_history.db database files.
+
+  The Linux sync_history.db database is typically stored in:
```
Comment (joachimmetz) [inline plaso/parsers/sqlite_plugins/dropbox.py:39]: I opt to put this into an artifact definition, if not already

**Review by joachimmetz — CHANGES_REQUESTED** [review]
Comments about some small cosmetic issues

**Comment by joachimmetz** [comment]
@sydp thx for putting these changes together, some comments about small cosmetic issues

File: data/formatters/generic.yaml (hunk 1 above)
Comment (sydp) [reply data/formatters/generic.yaml:102]: Would "File ID" be acceptable?  I note further above in the same file there is "Layer ID" and "Container ID".

File: plaso/parsers/sqlite_plugins/dropbox.py (hunk 2 above)
Comment (sydp) [reply plaso/parsers/sqlite_plugins/dropbox.py:20]: Yes, this appears to be a Dropbox generated value to uniquely reference a file.

```
File id (e.g. id:odTlUvbpIEFAAAAAAAAGOQ)
    - Unique file reference that persists if files are moved or renamed
    - Useful for apps referencing files that are moved around
    - Best choice for reliable, accurate reference to a specific file
 ```
https://developers.dropbox.com/dbx-file-access-guide

File: plaso/parsers/sqlite_plugins/dropbox.py (hunk 3 above)
Comment (sydp) [reply plaso/parsers/sqlite_plugins/dropbox.py:39]: Ack, already in the works.  Will submit a PR for the updated definition.

File: data/formatters/generic.yaml [hunk 4]
```diff
@@ -96,6 +96,16 @@ short_message:
 - 'Layer ID: {layer_id}'
 ---
 type: 'basic'
```
Comment (joachimmetz) [inline data/formatters/generic.yaml:98]: This should be conditional

**Review by joachimmetz — APPROVED** [review]
LGTM

**Pull request merged**